172599851 Sprites modificados y desarmar al jugador

Adrian-de-Lucas-Gomez opened the pull request:
Ahora al recibir un impacto de un objeto lanzable te podrá desarmar. Para ello debe superar una cierta velocidad para evitar que te pueda desarmar un objeto que estuviera flotando. Esa velocidad mínima está asociada a una constante.

También se modificaron los sprites del plátano para que estuviera mas en linea con el resto de las armas y también el color del router para mejorar su visibilidad en la sala de estar y la caldera (Los mapas mas oscuros).

Adrian-de-Lucas-Gomez committed:
Desarmar jugadores

Se ha creado una constante "DISARM_IMPACT" la cual marca el limite para poder ser golpeado por un arma y que no te pase nada. Si la velocidad que lleva lo sobrepasa te quedarás desarmado en el caso de que tuvieras un arma.
Por ahora solo te desarman las Weapon puras como la pelota saltarina por el problema de Weapon y sus clases hijas.

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172599851-SoltarArmasAlImpactar

Adrian-de-Lucas-Gomez committed:
Desarmar por colision de objetos lanzables

Para evitar problemas con los diferentes tipos de armas se ha optado por diferenciar los obketos si tienen el componente ThrownByPlayer ya que esto nos permite saber de forma implicita que se lanza y como sabemos los objetos que lanzamos (Los cuales todos son armas) y por lo tanto los que te pueden desarmar.

Se ha puesto un margen para la velocidad del impacto para evitar desarmarte al tocar con un arma que estaba estática.

Adrian-de-Lucas-Gomez committed:
Ajustes en sprites para coherencia

Simplemente para adaptarlos a los acpectos que tienen las armas cuando están flotando

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172599851-SoltarArmasAlImpactar

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172599851-SoltarArmasAlImpactar

Adrian-de-Lucas-Gomez committed:
Cambios al router

Solo un sprite mas claro para que se vea mejor

Adrian-de-Lucas-Gomez committed:
Eliminado en hueco que se dejaba al agarrase

Ya se queda bien pegadito a la pared. Lo he probado en la caldera que es la situación mas compleja.

Adrian-de-Lucas-Gomez committed:
Mejoras al platano

Si han refinado los sprites y se han hecho todos sobre el mismo diseño de Hands.

Tambien he quitado una cosa innecesaria que puse en el agarre

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172599851-SoltarArmasAlImpactar

Adrian-de-Lucas-Gomez committed:
Prueba para evitar clipping

Es una solucion y ayuda a la hora de hacer impulsos desde la pared pues de aleja un poco de ella.

Adrian-de-Lucas-Gomez committed:
Creo que ahora si (Los agarres)

Esto es un sufrimiento pero creo que ya estando en una sola pared no se puede clipear al otro lado. No he cambiado el grosor de las paredes pero como sea necesario se hará.

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172599851-SoltarArmasAlImpactar

Adrian-de-Lucas-Gomez committed:
Por Jesus bendito ya va bien

He dejado lo de que al impulsarte te separe un poco de la pared, ya no es necesario pero creo que beneficia a la movilidad evitando que roces con la pared.

Adrian-de-Lucas-Gomez committed:
Eliminada comprobacion con normales

Era demasiado restrictivo ya que hay paredes en diagonal

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172599851-SoltarArmasAlImpactar

Adrian-de-Lucas-Gomez committed:
Merge de las manitas

Adrian-de-Lucas-Gomez committed:
Ahora si que si

Creo que ya por fin he dado con la forma idonea para hacer los acercamientos a la pared. Basicamente usando las normales del plano pero en sentido contrario.

Adrian-de-Lucas-Gomez committed:
Mejoras del agarre

Se ha establecido un parametro de tolerancia de ángulos para evitar acercar demasiado al jugador si tiene un angulo cercano al paralelo a la pared

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172599851-SoltarArmasAlImpactar

Adrian-de-Lucas-Gomez committed:
Variables con nombres buenos

Que sino Cleon nos va a pegar

Adrian-de-Lucas-Gomez committed:
Comentado el ajuste de la posicion con los ángulos

eljorgemo commented on src/SourceFiles/Health.cpp:
```diff
@@ -105,8 +107,27 @@ void Health::onCollisionEnter(Collision* c)
 			PlayerData* playerWhoHitMe = nullptr;
-			//Si se impacta con un arma al umbral m�s alto de fuerza, se recibe su daño de impacto
+
+			//¿Es un objeto lanzable? (Por lo tanto puede desarmar)
```
En realidad, el que tenga el componente ThrownByPlayer no te garantiza que sea un arma. Lo que sí te lo garantiza es que tenga el componente Weapon, si no me equivoco (de ahí la importancia del static_cast que ha hecho will).
Puede haber un objeto que sea un arma pero no necesite llevar la puntuación del jugador que lo ha lanzado (si no ahora, igual en un futuro nos interesa)

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into SpritesModificadosYdesarmar

Adrian-de-Lucas-Gomez committed:
Cambio de comprobación al impactar

Ahora se mira directamente si es un arma evitando incoherencias y problemas en el futuro

JuanmaSalva reviewed (changes requested):
esto no es un cambio, me he equivocado

AdriAlvarez00 reviewed (changes requested):
Solo funciona con Health, no con Wallet.
Tambien estaría bien que se usase un metodo que deje el arma quieta, no que la tire

Adrian-de-Lucas-Gomez committed:
Ahora ya funciona con Wallet

Tambien he puesto optrop metodo que deja caer el arma con poquita velocidad para evitar que se vaya a la puta.

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into SpritesModificadosYdesarmar

Adrian-de-Lucas-Gomez committed:
Quitada la dedundancia

Ya vualve a haber solo un array para soltar armas

Adrian-de-Lucas-Gomez committed:
Listo

Ya no hay excepciones y todo va bien
